tests/testthat/test-examples.R

```diff
@@ -0,0 +1,90 @@
+# this test requires a `man` directory in the `tests/testthat` directory
+# (presumably symlinked to the package root `man` directory to avoid duplication)
+# this also requires `devtools::document()` to be run before running the tests
+
+rd_files <- function() {
+  list.files(testthat::test_path("man"), pattern = "\\.[Rr]d$", full.names = TRUE)
+}
+
+suppress_warnings <- function(expr, pattern = "*", ...) {
+  withCallingHandlers(expr, warning = function(w) {
+    if (grepl(pattern, conditionMessage(w))) {
+      invokeRestart("muffleWarning")
+    }
+  })
+}
+
+with_mocked_app_bindings <- function(code) {
+  shiny__shinyApp <- shiny::shinyApp # nolint object_name_linter.
+  mocked_shinyApp <- function(...) { # nolint object_name_linter.
+    print(shiny__shinyApp(...))
+  }
+
+  mocked_runApp <- function(x, ...) { # nolint object_name_linter.
+    app_driver <- shinytest2::AppDriver$new(
+      x,
+      shiny_args = list(...),
+      check_names = FALSE, # explicit check below
+      options = options() # https://github.com/rstudio/shinytest2/issues/377
+    )
+    on.exit(app_driver$stop(), add = TRUE)
```

Comment:
```suggestion
    on.exit(app_driver$stop(), add = TRUE)
    app_driver$wait_for_idle(timeout = 20000)
```

Without an explicit timeout the errors are not caught because of partial initialisation of the app.